ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/TimeServiceBackend.scala

```diff
@@ -12,52 +12,47 @@ import com.digitalasset.dec.DirectExecutionContext
 import scala.concurrent.Future
 
 trait TimeServiceBackend extends TimeProvider {
+  def clock = new TimeServiceBackendClock(this)
```

Comment:
Would it make sense to make this an `abstract class` and making `clock` a `val` rather than a `def`?

Reply:
Yes, good call.